crates/rslint_parser/src/syntax/stmt.rs

```diff
@@ -722,21 +731,24 @@ pub fn parse_while_statement(p: &mut Parser) -> ParsedSyntax {
 // const a = 5;
 // const { foo: [bar], baz } = {};
 // let foo = "lorem", bar = "ipsum", third = "value", fourth = 6;
-pub fn variable_declaration_statement(p: &mut Parser) -> CompletedMarker {
+pub fn variable_declaration_statement(p: &mut Parser) -> ParsedSyntax {
 	// test_err var_decl_err
 	// var a =;
 	// const a = 5 let b = 5;
 	let m = p.start();
 	let start = p.cur_tok().range.start;
 
-	variable_declaration(p, false);
+	parse_variable_declaration(p, false).or_missing(p);
 
 	semi(p, start..p.cur_tok().range.start);
```

Comment:
We need to check if we're indeed at a variable declaration statement and otherwise return absent. You can do this like this:

```suggestion
if let Present(declaration) = parse_variable_declaration(p, false) {
  let m = declaration.precede(p);
  semi(p, start..p.cur_tok().range.start);
  Present(m.complete(p, JS_VARIABLE_DECLARATION_STATEMENT))
} else {
  Absent
}
```